pages/api/inventory/CheckoutItems.js

```diff
@@ -0,0 +1,69 @@
+import firebase from '../../../firebase/clientApp'    
+import {validateFunc} from '../validate'
+
+/*
+* /api/inventory/CheckoutItems.js
+* req.body = {barcode1: quantity1, barcode2: quantity2, ...}
+*/
+
+function requireParams(body, res) {
+    /* require elements: array of elements {barcode: quantity} */
+
+    //require orders to have at least one item
+    if (body.length <= 0) {
+        res.json({error: "empty order"});
+        res.status(400);
+        return false;
+    }
+
+    return true;
+}
+
+export default async function(req,res) {  
+    
+    const token = req.headers.authorization
+    // console.log("TOKEN: ", token)
+    const allowed = await validateFunc(token)
+
+    if (!allowed) {
+        res.status(401);
+        res.json({error: "you are not authenticated to perform this action"})
+        return Promise.reject();
+    }
+
+    return new Promise((resolve, reject) => {
+        const {body} = req
+        console.log("(post-call) req: ", body);
+    
+        // verify parameters
+        let ok = requireParams(body, res);
+        if (!ok) {
+            return reject();
```

Comment:
Instead of returning `reject()` I think we have to set the status and json and then just `return` with no value. 
See [here](https://github.com/CalNourish/pantry-webapp/commit/0c89c8a43b74835e95961fdc4f2e698ac1a8ee9a#diff-ce5a43c9758eb18db8804c80cf23fc47837dde7b979b18570a784f8f02000864R17-R19).

Reply:
it seems like we still have to reject (the linked code was done in the `requireParams` function, and rejected later (line 80).
but i'll make the change to set the status and json!

Reply:
actually, nevermind! i do set the status and json in the requireParams function before it returns false, and then it rejects afterwards

Reply:
Ah you're right, nice!